frontend/packages/knative-plugin/src/components/knatify/CreateKnatifyPage.tsx

```diff
@@ -0,0 +1,111 @@
+import { ProjectModel } from '@console/internal/models';
+import {
+  LoadingBox,
+  Firehose,
+  FirehoseResource,
+  history,
+} from '@console/internal/components/utils';
+import NamespacedPage, {
+  NamespacedPageVariants,
+} from '@console/dev-console/src/components/NamespacedPage';
+import DeployImageForm from '@console/dev-console/src/components/import/DeployImageForm';
+import { handleRedirect } from '@console/dev-console/src/components/import/import-submit-utils';
+import { DeployImageFormData } from '@console/dev-console/src/components/import/import-types';
+import { deployValidationSchema } from '@console/dev-console/src/components/import/deployImage-validation-utils';
+import {
+  getInitialValuesKnatify,
+  knatifyResources,
+  getKnatifyWorkloadData,
+} from '../../utils/knatify-utils';
+
+const CreateKnatifyComponentLoader: React.FunctionComponent<any> = (props: any) => {
+  const { loaded, appName, namespace, handleSubmit, projects, workloadResource } = props;
+  const { t } = useTranslation();
+  return loaded && !_.isEmpty(workloadResource.data) ? (
+    <Formik
+      initialValues={getInitialValuesKnatify(
+        getKnatifyWorkloadData(workloadResource.data),
+        appName,
+        namespace,
+      )}
+      validationSchema={deployValidationSchema(t)}
+      onSubmit={handleSubmit}
+      onReset={history.goBack}
+    >
+      {(formikProps) => <DeployImageForm {...formikProps} projects={projects} isKnatify />}
+    </Formik>
+  ) : (
+    <LoadingBox />
+  );
+};
+
+type CreateKnatifyPageProps = RouteComponentProps<{ ns?: string }>;
+
+const CreateKnatifyPage: React.FunctionComponent<CreateKnatifyPageProps> = ({
+  match,
+  location,
+}) => {
+  const { t } = useTranslation();
+  const namespace = match.params.ns;
+  const queryParams = new URLSearchParams(location.search);
+  const kind = queryParams.get('kind');
+  const appName = queryParams.get('name');
+  const [perspective] = useActivePerspective();
+  const perspectiveExtensions = useExtensions<Perspective>(isPerspective);
+
+  const appResources: FirehoseResource[] = [
+    {
+      kind: ProjectModel.kind,
+      prop: 'projects',
+      isList: true,
+    },
+    {
+      kind,
+      prop: 'workloadResource',
+      name: appName,
+      namespace,
+      optional: true,
+    },
+  ];
+
+  const handleSubmit = (
+    values: DeployImageFormData,
+    helpers: FormikHelpers<DeployImageFormData>,
+  ) => {
+    return knatifyResources(values)
+      .then(() => {
+        helpers.setStatus({ submitError: '' });
+        handleRedirect(namespace, perspective, perspectiveExtensions);
+      })
+      .catch((err) => {
+        helpers.setStatus({ submitError: err.message });
+      });
+  };
+
+  return (
+    <NamespacedPage disabled variant={NamespacedPageVariants.light}>
+      <Helmet>
+        <title>{t('knative-plugin~Knatify')}</title>
+      </Helmet>
+      <div className="co-m-pane__body">
+        <Firehose resources={appResources}>
```

Comment:
Why not use the hook? `useK8sWatchResources`

Reply:
upated